elrond-wasm/src/contract_base/wrappers/blockchain_wrapper.rs

```diff
@@ -81,6 +81,12 @@ where
         }
     }
 
+    pub fn check_caller_is_user_account(&self) {
```

Comment:
can be done more efficiently by going directly to the API:
```
A::blockchain_api_impl().load_caller_managed(const_handles::MB_TEMP_1);
A::blockchain_api_impl().is_smart_contract(const_handles::MB_TEMP_1)
```
Avoids creating a new handle